\section*{Background}
The study of mutational signatures — the idiosyncratic patterns of somatic alterations left by mutagenic processes on the genomes of cells — has acquired considerable scientific prominence in recent years, particularly within the field of cancer genomics \cite{Alexandrov2014, Petljak2016}. The identification of mutational signatures and the quantification of their activities in a genome offer valuable insight into the molecular processes that operate on the DNA of cancer and normal cells. These may include endogenous mutagenic processes, such as those arising from deficiencies in DNA repair pathways, and exposures to exogenous carcinogens, such as tobacco smoke \cite{Petljak2016}.

Most analyses of mutational signatures so far have focused on patterns of mutations defined over a categorisation of single-nucleotide variants (SNVs). Specifically, 96 trinucleotide mutation categories are defined for SNVs based on base substitution type (6 categories) and trinucleotide sequence context (i.e. the bases immediately 5' and 3' of the mutated base; 16 categories). Base substitution types are collapsed so that only pyrimidine reference bases (cytosine or thymine) are considered. (For instance, guanine-to-adenine and cytosine-to-thymine mutations represent the same base change on opposite strands, so both can be expressed as cytosine-to-thymine, or C$>$T.) A vector of mutation counts or mutation probabilities across these 96 mutation categories is known as a mutational catalogue, or spectrum \cite{Alexandrov2014, Alexandrov2013}.

A considerable number of software tools for mutational signature analysis have entered the scientific literature over the last five years \cite{Baez-Ortega2017}. Perhaps most prominent among these is the Wellcome Trust Sanger Institute Mutational Signature Framework \cite{AlexandrovCellRep2013}. This was the first published method to perform inference of mutational signatures \textit{de novo} (also known as signature extraction), and implements a non-negative matrix factorization (NMF) algorithm that has been repeatedly applied to blind source separation problems in biology and other fields \cite{Devarajan2008}. An alternative to this NMF approach is the probabilistic model introduced by the EMu software \cite{Fischer2013}, which implements an expectation–maximisation (EM) algorithm to perform maximum-likelihood estimation using a Poisson mutational model, and is able to account for differences in the opportunity for mutations of each class to occur in the sequence. Later tools have aimed to estimate the proportions in which a set of predefined mutational signatures are present in a collection of catalogues (a problem known as signature fitting). Notably, signature fitting does not share the considerable sample-size requirements of signature extraction methods, and is applicable even to individual mutational catalogues. However, recurrent drawbacks of algorithms for mutational signature analysis include data overfitting (overestimating the number of mutational processes), elevated computational cost, limited ranges of accepted mutation categories, poor integration with existing statistical programming frameworks and incomplete documentation.

Here we present sigfit, a powerful, flexible and user-friendly software package for performing mutational signature analysis in the R programming language. It provides both NMF- and EMu-inspired Bayesian statistical models for signature extraction and signature fitting, as well as novel statistical models that enable simultaneous fitting and extraction of signatures, aimed at detection of rare or previously undescribed mutational patterns. These models can be seamlessly applied to any kind of categorised mutation data, including short insertions and deletions (indels), dinucleotide variants, large structural variants and copy number alterations. The package also incorporates functions for production of publication-quality graphics of input and output data, and extensive user documentation featuring usage examples and test data sets.
    